### PR TITLE
Add instructions for migration from NFS to StorageClass

### PR DESCRIPTION
**Pull request by flin-8**
[sc-76947]
Add a section for NFS -> StorageClass migration

CleanShot 2024-05-01 at 11 06 37@2x

### CODE REVIEW

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md [hunk 1]
```diff
@@ -52,3 +52,27 @@ If you manage your own cluster and don’t have offerings from cloud providers a
...
+1. Take note of the current Helm release and Chart version for your Kubernetes agent by running the following command:
+   ```bash
+   helm list --namespace octopus-agent-nfs-to-pv
+   ```
+1. The output should look like this:
+   - In this example, the release name is `nfs-to-pv` while the chart version is `1.0.1`
+   :::figure
+   ![Helm list command](/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/kubernetes-agent-helm-list.png)
+   :::
+1. Run the following command (substitute with your own values):
```
Comment (flin-8) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md:71]: This part looks a bit busy, any suggestions?

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ If you have a use case that can’t tolerate occasional deployment failures, it
 
 If you need a more reliable storage solution, then you can specify your own `StorageClass`. This `StorageClass` must be capable of `ReadWriteMany` (also known as `RWX`) access mode. 
 
-Many managed Kubernetes offerings will have offerings that require little effort to use. These will be a “provisioner” (named as such as they “provision” storage for a storage class), which you can then tie to a storage class. Some examples are listed below:
+Many managed Kubernetes offerings will have provide storage that require little effort to set up. These will be a “provisioner” (named as such as they “provision” storage for a `StorageClass`), which you can then tie to a `StorageClass`. Some examples are listed below:
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md:43]:
```suggestion
Many managed Kubernetes offerings will provide storage that require little effort to set up. These will be a “provisioner” (named as such as they “provision” storage for a `StorageClass`), which you can then tie to a `StorageClass`. Some examples are listed below:
```

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md [hunk 3]
```diff
@@ -52,3 +52,27 @@ If you manage your own cluster and don’t have offerings from cloud providers a
 - [Longhorn](https://longhorn.io/)
 - [Rook (CephFS)](https://rook.io/)
 - [GlusterFS](https://www.gluster.org/)
+
+## Migrating from NFS storage to a custom StorageClass
+
+If you installed the Kubernetes agent using the default NFS storage, and want to use a custom `StorageClass` instead, simply rerun the installation Helm command with specified values for `persistence.storageClassName` and `persistence.size`.
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md:58]:
```suggestion
If you installed the Kubernetes agent using the default NFS storage, and want to change to custom `StorageClass` instead, simply rerun the installation Helm command with specified values for `persistence.storageClassName` and `persistence.size`.
```

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md (hunk 1 above)
Comment (APErebus) [reply src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md:71]: There is a code block component that might make this easier to layout?

Example
https://octopus.com/docs/deployments/custom-scripts#scripts-that-block-deployments

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md (hunk 3 above)
Comment (APErebus) [reply src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md:58]: Technically they don't need to set the size here unless they want to increase the volume size

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md [hunk 4]
```diff
@@ -52,3 +52,27 @@ If you manage your own cluster and don’t have offerings from cloud providers a
...
+
+The following steps assume your Kubernetes agent is in the `octopus-agent-nfs-to-pv` namespace:
+
+1. Take note of the current Helm release and Chart version for your Kubernetes agent by running the following command:
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md:62]:
```suggestion
1. Take note of the current Helm release name and Chart version for your Kubernetes agent by running the following command:
```

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md (hunk 1 above)
Comment (flin-8) [reply src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md]: I split the steps out into headings - which made it nicer

**Review by APErebus — APPROVED** [review]
very nice!

**Pull request merged**